### PR TITLE
Improve deploy.sh by adding documents and post-deployment summary

### PR DESCRIPTION
**Pull request by mb1896**
Output after executing the script:

```
############################
#                          #
# IPC deployment ready! 🚀 #
#                          #
############################
Subnet ID:
/r314159/t410ff7zdb3ytj4zso5boanhpvyiuupqatwjuply4laa

ETH API:
http://localhost:8545
http://localhost:8645
http://localhost:8745

Accounts:
t410fwt3jxcv4lm5h7gyikzqvac4jriu7vptyoo5c7iq: 1000000000000000000 coin units
t410f566zdxg72onhviwlmvtfhcwfpsh6cyethhfojia: 1000000000000000000 coin units
t410f6fq3uxtr5jslgtuvvbqphmetijohzknp4n5ycrq: 1000000000000000000 coin units

Private keys (hex ready to import in MetaMask):
73d69defa6b5e5f6fb7dcf5ddba6fc6f87b579ed9d77be76e9c6dc75bedc6fc6f7d5f6daf78edaf39e9a6b46b77f7e35
e1d73c73de9cdded5c7dbebde9ce9a75a7bdebbd1df39738d5aedae5f7f6e3bf1fd39eb77b7e9bd1c7f6d7bdfde5bd34
f3de9e79debcedc6f9d1dddee5c6f8edcf5e79a71cf38d1aef5d9bf5df37d9ee5cd5fe3c6b86f56fbdfadb7f3bdbc79f

Chain ID:
2833157499179602

Fendermint API:
http://localhost:26658

CometBFT API:
http://localhost:26657
```

Closes ENG-715

### CODE REVIEW

File: .github/workflows/deploy-to-dedicated-host.yaml [hunk 1]
```diff
@@ -2,6 +2,7 @@ name: Deploy to calibration net on a dedicated server
 
 on:
   workflow_dispatch:
+  pull_request:
```
Comment (mb1896) [inline .github/workflows/deploy-to-dedicated-host.yaml:5]: Will remove before merge

File: scripts/deploy_subnet_under_calibration_net/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+## Deploy to remote host
+We currently have a [Github workflow](https://github.com/consensus-shipyard/ipc/actions/workflows/deploy-to-dedicated-host.yaml) to deploy IPC infra to a dedicated host. You can go to the workflow page and click `Run workflow` button on the top right corner to initiate a deployment.
+
+## Deploy to local machine
+The same `deploy.sh` script can also be used to deploy locally. This is more or less equivalent to following [quickstart-calibration.md](https://github.com/consensus-shipyard/ipc/blob/main/docs/ipc/quickstart-calibration.md), but much more automated.
+
+To run this script locally, you need to first manually prepare the environment and files.
+
+1. Make sure you have your ipc repo located at $HOME/ipc.
+2. Follow Step 2 and Step 3 in [Github workflow](https://github.com/consensus-shipyard/ipc/actions/workflows/deploy-to-dedicated-host.yaml) to prepare ipc config file and wallets. Remember to go to Calibration faucet to fund all of your addresses.
```
Comment (maciejwitowski) [inline scripts/deploy_subnet_under_calibration_net/README.md:12]:
```suggestion
2. Follow Step 2 and Step 3 in [Github workflow](https://github.com/consensus-shipyard/ipc/actions/workflows/deploy-to-dedicated-host.yaml) to prepare ipc config file and wallets. Remember to go to [Calibration faucet](https://faucet.calibration.fildev.network/funds.html) to fund all of your addresses.
```

File: scripts/deploy_subnet_under_calibration_net/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+3. Run `bash deploy.sh local` to deploy IPC locally.
+
+Please also notice that
+1. The automated dependency installation isn't guarantee to work 100% time. If you encountered any dependency installation issue, please refer to the script and retry. Usually you can resolve the issues by creating a new terminal, sourcing `~/.bash.rc`, etc.
+2. Depends on the RPC endpoint's quality of service for the calibration net, your command may or may not succeed when interacting with the RPC endpoint. Sometimes you will get rate limited, or other weird errors. In that case, you can choose a different calibration provider URL (TODO: Add link) to replace the value of `RPC_URL` variable in the script, then retry it.
```
Comment (maciejwitowski) [inline scripts/deploy_subnet_under_calibration_net/README.md:17]:
```suggestion
2. Depends on the RPC endpoint's quality of service for the calibration net, your command may or may not succeed when interacting with the RPC endpoint. Sometimes you will get rate limited. In that case, you can choose a different calibration provider URL from [Chainlist](https://chainlist.org/?search=calibration&testnets=true) to replace the value of `RPC_URL` variable in the script, then retry it.
```

**Review by maciejwitowski — COMMENTED** [review]
Hey @mb1896, the workflow returned error when I ran it https://github.com/consensus-shipyard/ipc/actions/runs/7904440787/job/21574711673

**Comment by maciejwitowski** [comment]
One q: do I remember correctly that you said the local flow doesn't work for Mac? If so it'd be great to make it clear here.

**Comment by mb1896** [comment]
> Hey @mb1896, the workflow returned error when I ran it https://github.com/consensus-shipyard/ipc/actions/runs/7904440787/job/21574711673

Seems like manual triggered run doesn't give me {{ head.ref }} as expected. Let me try to fix it.

File: scripts/deploy_subnet_under_calibration_net/README.md (hunk 2 above)
Comment (mb1896) [reply scripts/deploy_subnet_under_calibration_net/README.md]: Thanks! fixed

File: scripts/deploy_subnet_under_calibration_net/README.md (hunk 3 above)
Comment (mb1896) [reply scripts/deploy_subnet_under_calibration_net/README.md]: Thanks! fixed

**Comment by mb1896** [comment]
> One q: do I remember correctly that you said the local flow doesn't work for Mac? If so it'd be great to make it clear here.

Added related note for running on Mac.

**Review by maciejwitowski — APPROVED** [review]
LGTM, thank you!

**Pull request merged**